File: Python/05_phase/phaserepair.c
// NOTE: it is recommended to use this even if you don't understand the following code.

#include <assert.h>
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

#define MAXB 200000
#define MAXN 200000

int B, M, N, i;
int A[MAXN];
int R[MAXB];

int main() {
    // uncomment the two following lines if you want to read/write from files
    // freopen("input.txt", "r", stdin);
    // freopen("output.txt", "w", stdout);

    assert(2 == scanf("%d%d", &N, &M));

    for (i = 0; i < N; ++i)
        assert(1 == scanf("%d", &A[i]));

    B = 0;


    // INSERT YOUR CODE HERE


    printf("%d\n", B);
    for (i = 0; i < B; ++i)
        printf("%d ", R[i]);
    printf("\n");

    return 0;
}

File: Python/05_phase/phaserepair.py
#!/usr/bin/env python3
# NOTE: it is recommended to use this even if you don't understand the following code.

import sys

# uncomment the two following lines if you want to read/write from files
sys.stdin = open('input1.txt')
sys.stdout = open('output1.txt', 'w')

N, M = map(int, input().strip().split())

A = list(map(int, input().strip().split()))

B = 0
R = [0 for i in range(B)]


# INSERT YOUR CODE HERE
A.sort()
max_dist = 0 

if(A[0] == A[N - 1]):
    B = 0
else:
    for i, c in enumerate(A):
        if abs(c[i] - c[i+1]) > max_dist:
            max_dist = abs(c[i] - c[i+1])

    if(M == 2):
        B = 1

    elif((M - 1) == max_dist):
        B = max_dist - 1
        for i in range(max_dist - 2):
                R.append(M - 1 -i)
    R.sort()
    R.append(0)



print(B)
for i in range(B):
    print(R[i], end=' ')
print()

sys.stdout.close()
